core/src/main/java/org/infinispan/health/impl/CacheHealthImpl.java

```diff
@@ -9,36 +8,36 @@
 
 public class CacheHealthImpl implements CacheHealth {
 
-    private final AdvancedCache<?, ?> cache;
+   private final Cache<?, ?> cache;
 
-    public CacheHealthImpl(Cache<?, ?> cache) {
-        this.cache = cache.getAdvancedCache();
-    }
+   public CacheHealthImpl(Cache<?, ?> cache) {
+      this.cache = cache;
+   }
 
-    @Override
-    public String getCacheName() {
-        return cache.getName();
-    }
+   @Override
+   public String getCacheName() {
+      return cache.getName();
+   }
 
-    @Override
-    public HealthStatus getStatus() {
-        if (!isComponentHealthy() || cache.getAvailability() == AvailabilityMode.DEGRADED_MODE) {
-            return HealthStatus.UNHEALTHY;
-        }
-        DistributionManager distributionManager = SecurityActions.getDistributionManager(cache);
-        if (distributionManager != null && distributionManager.isRehashInProgress()) {
-            return HealthStatus.REBALANCING;
-        }
-        return HealthStatus.HEALTHY;
-    }
+   @Override
+   public HealthStatus getStatus() {
+      if (!isComponentHealthy() || cache.getAdvancedCache().getAvailability() == AvailabilityMode.DEGRADED_MODE) {
+         return HealthStatus.UNHEALTHY;
+      }
+      DistributionManager distributionManager = SecurityActions.getDistributionManager(cache);
+      if (distributionManager != null && distributionManager.isRehashInProgress()) {
+         return HealthStatus.REBALANCING;
+      }
+      return HealthStatus.HEALTHY;
+   }
 
-    private boolean isComponentHealthy() {
-        switch (cache.getStatus()) {
-            case INSTANTIATED:
-            case RUNNING:
-                return true;
-            default:
-                return false;
-        }
-    }
+   private boolean isComponentHealthy() {
+      switch (cache.getAdvancedCache().getStatus()) {
```

Comment:
And here?

Reply:
same reason as before, it was `advancedCache